### ISSUE TITLE
Update endpoint to retrieve and send Claims Evidence Documents from Lighthouse source

### ISSUE THREAD

**Issue by kellylein** [department-of-veterans-affairs/va-mobile-app#4746]
There is a new Lighthouse API that we need to use instead of EVSS to pull and send claim evidence documents.  The VA.gov Benefits CLaims and Appeals Status team is building an integration to the new LH Benefits Documents API so we should utilize their integration.

Endpoints:
/v0/claim/{id}/documents
/v0/claim/{id}/documents/multi-image

AC:

 - [ ] Update our endpoint(s) to send Claims Evidence documents to the new VA.gov integration to LH
 - [ ] Ensure all data is returned to the app in the same format

**Comment by aherzberg** [comment]
still blocked by https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/6395. see ticket for updates

**Comment by aherzberg** [comment]
issue has been solved causing failures to the LH endpoint. Still some blockers on this as follows: 

1. AWS S3 lighthouse bucket has still not been created. I talked with Kshitiz Shrestha on devops support to get update today and he said "For that s3 I am planning to have it created separately rather than running down rabbit hole. I can try to complete it by tomorrow"
2. the manifests and devops changes we made to account for the new bucket was only for staging (devops prefers us to only change one env at a time). Both repos need to be updated for prod too. 
3. Though the async job is no longer failing, when going through the mobile endpoint, the document is not showing on the claims `show` endpoint like it is when we hit the LH endpoint directly. This may be a postman issue but will require more investigation. 

I was told to focus on current sprint tickets for now but will come back to this.

**Comment by aherzberg** [comment]
item 3 from above is now fixed.

**Comment by aherzberg** [comment]
Ready to test. make sure that the user has the `mobile_lighthouse_document_upload` flag enabled. right now 54 has it enabled. expected behavior is that nothing is changed between evss and LH

**Comment by narin** [comment]
@aherzberg I'm seeing some unexpected results after looking in VBMS. From @TKDickson's overview, once files are uploaded, the "Received date" in the tracked items table should be populated, but none of the items in any of the 3 claims are.  Also, there were only 5 files in the eFolder that were uploaded on 9/6 even though there should be 27.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcGVRIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--3670cd808f2298e1c1742d7678eab6e5198a0029/image.png)
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcGlRIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ed908f8a1e59c0331450875f41f7333b64883621/image.png)

**Comment by aherzberg** [comment]
@narin Interesting. this was all for user vets.gov.user+54@gmail.com? and it was claim  600438777? I'll start a conversation with Lighthouse to make sure they're hooked into VBMS as expected on staging. was there any types of files that seemed to be the ones left out? Looks like at least some multifile uploads were successful and some single file were based on the document titles

**Comment by narin** [comment]
@aherzberg Yup, vets.gov.user+54@gmail.com. The uploads were across 3 claims:

- 600438772
- 600438777
- 600438778

@rbontrager based on the screenshots I sent you, can you comment Andrew's last question about the missing files?

**Comment by aherzberg** [comment]
Sounds good. looking into this.

**Comment by kellylein** [comment]
@jperk51 to check in with @aherzberg once he is back to see how to get this moving.

**Comment by bischoffa** [comment]
Adding Sprint 81 as the ticket was not closed out in 80. Carryover work.

**Comment by aherzberg** [comment]
pending PR merge https://github.com/department-of-veterans-affairs/vets-api/pull/13827. There's been some changes to codeowners in vets-api, requiring more reviews from a lighthouse team. I've pinged them again today to get a review. Once reviewed, i can merge it and we can test the fix. https://dsva.slack.com/archives/C02CQP3RFFX/p1695829165754829

**Comment by aherzberg** [comment]
PR has been merged. Staging is having trouble syncing due to some devops change so hasn't deployed the change yet. Looks to be a known issue so going to keep an eye on it and test once it's been deployed

**Comment by aherzberg** [comment]
staging still isn't deploying. created ticket: https://dsva.slack.com/archives/CBU0KDSB1/p1696008553514249

**Comment by aherzberg** [comment]
Non-pdf files are now being uploaded but they are showing their filetype as pdf even though what we send is not a pdf. Doing some additional investigation

**Comment by aherzberg** [comment]
Turns out that was intended behavior. @rbontrager @kellylein Ready for QA testing again!

**Comment by narin** [comment]
@aherzberg: 

@rbontrager uploaded 10 more files today (10/4).  A couple interesting results:
1. The eDocuments folder is showing 20 files uploaded even though Rachael uploaded 10 only. On a positive note, it seems all the files she uploaded are present
2. The 'Received Date' is still not being populated in the Tracked Items tab

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0VHQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--83544457fc5058e0953dc2e26a1419187add9bdc/Screenshot%202023-10-04%20at%204.26.08%20PM.png)

**Comment by aherzberg** [comment]
Got it. I brought it up with LH to see if it something on their side or is expected behavior. We don't pass an expected date so hopefully shouldn't be an issue on our side.

**Comment by bischoffa** [comment]
Carryover from Sprint 81 into Sprint 82

**Comment by TKDickson** [comment]
@narin and @rbontrager I am taking this over, @aherzberg I will get back to you about the file duplication question (hopefully today!)

**Comment by TKDickson** [comment]
Figured out the duplication problem (hat tip to @rbontrager for the insight!)

Something about how file requests are being returned is no longer marking them as submitted/no longer needed (some relevant FE code snippets below), so steps to reproduce this issue are:
Veteran submits file for request A.
Request A is marked as submitted.
Veteran backs out of the claim, then re-enters - Request A is no longer marked as submitted.
Veteran submits ANOTHER file for request A.
Repeat ad nauseum.

Makes me wonder if the [“date received” not being populated](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/4746#issuecomment-1747590568) that Narin had pointed out/Andrew asked about earlier is a “should be fixed before go live” issue, related to this problem.

```
export const needItemsFromVet = (attributes: ClaimAttributesData): boolean => {
  return !attributes.decisionLetterSent && attributes.open && attributes.documentsNeeded && numberOfItemsNeedingAttentionFromVet(attributes.eventsTimeline) > 0
}

export type ClaimAttributesData = {
  dateFiled: string
  minEstDate: string | null
  maxEstDate: string | null
  phaseChangeDate: string | null
  open: boolean
  waiverSubmitted: boolean
  documentsNeeded: boolean
  developmentLetterSent: boolean
  decisionLetterSent: boolean
  phase: number
  everPhaseBack: boolean
  currentPhaseBack: boolean
  requestedDecision: boolean
  claimType: string
  updatedAt: string
  contentionList: Array<string>
  vaRepresentative: string
  eventsTimeline: Array<ClaimEventData>
}
```

**Comment by TKDickson** [comment]
Waiting [for Lighthouse fixes for issues above](https://dsva.slack.com/archives/C018V2JCWRJ/p1697740943337839?thread_ts=1697047775.219099&cid=C018V2JCWRJ)

**Comment by TKDickson** [comment]
Technically, all of our work is done (and we shouldn't _need_ to make backend changes after upstream fixes are in) so splitting out a 3rd party bug ticket for the stuff above (that QA will need to do testing on when it's resolved)

**Comment by kellylein** [comment]
Bug related to this ticket: #7119
